Enhance RAG Module with Post-Processor Support and Jina Reranking

asterixcapri opened the pull request:
These updates improve document retrieval accuracy and flexibility in the RAG module.

Key Changes:
- Introduced `PostProcessorInterface` and integrated post-processor handling in RAG.
- Added `JinaRerankerPostProcessor` to reorder documents using Jina's API.
- Implemented `getRetrievedDocuments()` method for accessing retrieved documents.

Please review and provide feedback.

ilvalerione commented:
- You introduced a new interface but the Jina concrete implementation does not implement this interface
- There are no unit tests

What other use cases are there for post-processing other than re-ranking?

asterixcapri commented:
Fixed missing interface (`JinaRerankerPostProcessor` now implements `PostProcessorInterface`) and I added unit tests.
CI is failing on docker/login-action@v3 (username/password required). I'm not sure why.

ilvalerione commented:
The workflow failed in your branch because it cannot access to our internal GitHub secrets.

Thank you for working on it, it seems a nice addition. I released this implementation as part of the [1.9.17](https://github.com/inspector-apm/neuron-ai/releases/tag/1.9.17) version.

I changed it a bit and added observability events so you can get the post process execution and results insights into the Inspector dashboard.

Thank you for the contribution. It's time now for the documentation and spread the word across our network. Feel free to post about it on the forum: https://github.com/inspector-apm/neuron-ai/discussions

More people use it more improvements will come.

asterixcapri commented:
Thanks! Give me just a few minutes, I've got a last-minute fix coming. Sorry
